Sources/SwiftDocC/Model/Identifier.swift

```diff
@@ -131,9 +131,22 @@ public struct ResolvedTopicReference: Hashable, Codable, Equatable, CustomString
     }
     
     public init(bundleIdentifier: String, path: String, fragment: String? = nil, sourceLanguages: Set<SourceLanguage>) {
+        self.init(
```

Comment:
Beyond the copy-on-write change, which by itself provides a net gain in performance, I would suggest keeping the changes in this type to separate PR to manage risk (e.g., if we need to revert this PR for whatever reason, the additional improvements would not be reverted).

Reply:
This is a good point. I've rebased the commits in this PR to 5 isolated commits and will plan on a rebase and merge instead of a squash so that we can revert individual parts of this PR if necessary.